src/System.IO/src/System/IO/TextWriter.cs

```diff
@@ -597,5 +598,332 @@ public override void Write(char value)
             {
             }
         }
+
+#if netstandard17
+        public static TextWriter Synchronized(TextWriter writer)
+        {
+            if (writer == null)
+                throw new ArgumentNullException(nameof(writer));
+
+            if (writer is SyncTextWriter)
+                return writer;
+
+            return new SyncTextWriter(writer);
+        }
+
+        [Serializable]
+        internal sealed class SyncTextWriter : TextWriter, IDisposable
+        {
+            private TextWriter _out;
+
+            internal SyncTextWriter(TextWriter t) : base(t.FormatProvider)
+            {
+                _out = t;
+            }
+
+            public override Encoding Encoding
+            {
+                get { return _out.Encoding; }
+            }
+
+            public override IFormatProvider FormatProvider
+            {
+                get { return _out.FormatProvider; }
+            }
```

Comment:
Nit: if you wanted, lots of these members (methods and properties) are one-liners and could be simplified with expression-bodied members, e.g.
```C#
public override Encoding Encoding => _out.Encoding;

public override IFormatProvider FormatProvider => _out.FormatProvider;

[MethodImpl(MethodImplOptions.Synchronized)]
public override void Close() => _out.Close();

... // etc.
```